src/System.Data.Common/src/System/Data/LinqDataView.cs

```diff
@@ -0,0 +1,298 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Data;
+using System.Diagnostics;
+
+namespace System.Data
+{
+    /// <summary>
+    /// Represents a bindable, queryable DataView of DataRow, that can be created from from LINQ queries over DataTable
+    /// and from DataTable. 
+    /// </summary>
+    internal class LinqDataView : DataView, IBindingList, IBindingListView
+    {
+        /// <summary>
+        /// A Comparer that compares a Key and a Row.
+        /// </summary>
+        internal Func<object, DataRow, int> comparerKeyRow;  // comparer for DataView.Find(..
+
+        /// <summary>
+        /// Builds the sort expression in case multiple selector/comparers are added
+        /// </summary>
+        internal readonly SortExpressionBuilder<DataRow> sortExpressionBuilder;
+
+        /// <summary>
+        /// Constructs a LinkDataView and its parent DataView.
+        /// Does not create index on the DataView since filter and sort expressions are not yet provided.
+        /// </summary>
+        /// <param name="table">The input table from which LinkDataView is to be created.</param>
+        internal LinqDataView(DataTable table, SortExpressionBuilder<DataRow> sortExpressionBuilder)
+            : base(table)
+        {
+            Debug.Assert(table != null, "null DataTable");
+            this.sortExpressionBuilder = sortExpressionBuilder ?? new SortExpressionBuilder<DataRow>();
+        }
+
+
+        //I have two forms of predicate because I need to pass in null if predicate is null. Otherwise I need to convert it into delegate and pass it into
```

Comment:
Not sure whether this should be in the `<summary>` ? It is empty- you could delete it ?